Assets/Scripts/Player/Animation.cs

```diff
@@ -0,0 +1,63 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Animation : MonoBehaviour
+{
+    public float movementSpeed = 3f;
+    public float sprintingSpeed = 6f;
+    public float superSpeed = 20f;
+    private float actuallSpeed;
+    public Vector2 movement;
+    public Rigidbody2D rigidbody;
+
+
+    public Animator playerAnimator;
+    public float hf = 0.0f;
+    public float vf = 0.0f;
+
+    void Start()
+    {
+        rigidbody = this.GetComponent<Rigidbody2D>();
+        actuallSpeed = movementSpeed;
+    }
+
+    void Update()
+    {
+        movement.x = Input.GetAxisRaw("Horizontal");
+        movement.y = Input.GetAxisRaw("Vertical");
+        movement = movement.normalized;
+
+        if (Input.GetKeyDown(KeyCode.LeftShift) && actuallSpeed == movementSpeed)
+        {
+            actuallSpeed = actuallSpeed + sprintingSpeed;
+        }
+        if (Input.GetKeyUp(KeyCode.LeftShift) && actuallSpeed == (movementSpeed + sprintingSpeed))
+        {
+            actuallSpeed = actuallSpeed - sprintingSpeed;
+        }
+
+        if (Input.GetKeyDown("l") && actuallSpeed == movementSpeed)
+        {
+            actuallSpeed = actuallSpeed + superSpeed;
+        }
+        if (Input.GetKeyUp("l") && actuallSpeed == (movementSpeed + superSpeed))
+        {
+            actuallSpeed = actuallSpeed - superSpeed;
+        }
```

Comment:
Also, shouldn't you assign the speed to a constant value (i.e. `currentSpeed = superSpeed`? Otherwise you **will** get problems the moment you can i.e. both have super and sprinting speed (as is currently allowed, if I see that correctly.))

Reply:
Because of the `actuallSpeed == movementSpeed` in line 40 and 31 if you're sprinting you can't get super speed and vice versa. This is because if you're sprinting `actuallSpeed = actuallSpeed + sprintingSpeed` and if you have super speed `actuallSpeed = actuallSpeed + superSpeed`